### PR TITLE
Feature number-count animation for DORA-cards

### PR DESCRIPTION
**Pull request by xyfer17**
## Linked Issue(s)

#385 

## Acceptance Criteria fulfillment

- [X] Task 1
- [X] Task 2
- [X] Task 3

## Proposed changes (including videos or screenshots)

https://github.com/user-attachments/assets/b25c3683-2748-4c86-a503-e77592d1df96

## Further comments

As per issue #385 , have Implement a number-count animation for DORA-cards using custom hook logic.

### CODE REVIEW

**Comment by jayantbh** [comment]
You can undo the axios change. Please rebase onto main.

**Comment by xyfer17** [comment]
> You can undo the axios change. Please rebase onto main.

done, pushed

File: web-server/src/hooks/useCountUp.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export const useCountUp = (
+  targetValue: number,
+  duration: number = 1000,
+  decimalPlaces: number = 0
+): number => {
+  const [count, setCount] = useState<number>(0);
+
+  useEffect(() => {
+    let start = 0;
+    const increment = targetValue / (duration / 16);
```
Comment (jayantbh) [inline web-server/src/hooks/useCountUp.ts:12]: Why 16? Prefer not using magic numbers.
Fixing them can be as simple as declaring them as a constant somewhere.

In this case, right outside the hook might be fine.

File: web-server/src/content/DoraMetrics/DoraCards/WeeklyDeliveryVolumeCard.tsx [hunk 2]
```diff
@@ -55,6 +56,11 @@ export const WeeklyDeliveryVolumeCard = () => {
   const dateRangeLabel = useCurrentDateRangeLabel();
   const deploymentFrequencyProps = useAvgIntervalBasedDeploymentFrequency();
 
+  const deploymentFrequencyCount = useCountUp(
+    deploymentFrequencyProps.count || 0,
+    1500
```
Comment (jayantbh) [inline web-server/src/content/DoraMetrics/DoraCards/WeeklyDeliveryVolumeCard.tsx:61]: Same thing here. Magic numbers. Ref: The other comment.

File: web-server/src/content/DoraMetrics/DoraCards/MeanTimeToRestoreCard.tsx [hunk 3]
```diff
@@ -76,6 +77,11 @@ export const MeanTimeToRestoreCard = () => {
     ]
   );
 
+  const meanTimeToRestoreCount = useCountUp(
+    meanTimeToRestoreProps.count || 0,
+    1500
```
Comment (jayantbh) [inline web-server/src/content/DoraMetrics/DoraCards/MeanTimeToRestoreCard.tsx:82]: Same.

File: web-server/src/content/DoraMetrics/DoraCards/ChangeFailureRateCard.tsx [hunk 4]
```diff
@@ -80,6 +81,12 @@ export const ChangeFailureRateCard = () => {
         .incident_count
   );
 
+  const changeFailureRateCount = useCountUp(
+    changeFailureRateProps.count || 0,
+    1500,
```
Comment (jayantbh) [inline web-server/src/content/DoraMetrics/DoraCards/ChangeFailureRateCard.tsx:86]: If you intend to use 1500 as the duration everywhere, and for now this hook isn't used anywhere else...

Why not just use 1500 as the default argument?

**Review by jayantbh — CHANGES_REQUESTED** [review]
This is 95% done. Nice work!

File: web-server/src/content/DoraMetrics/DoraCards/WeeklyDeliveryVolumeCard.tsx (hunk 2 above)
Comment (xyfer17) [reply web-server/src/content/DoraMetrics/DoraCards/WeeklyDeliveryVolumeCard.tsx]: fixed

File: web-server/src/content/DoraMetrics/DoraCards/MeanTimeToRestoreCard.tsx (hunk 3 above)
Comment (xyfer17) [reply web-server/src/content/DoraMetrics/DoraCards/MeanTimeToRestoreCard.tsx]: fixed

File: web-server/src/content/DoraMetrics/DoraCards/ChangeFailureRateCard.tsx (hunk 4 above)
Comment (xyfer17) [reply web-server/src/content/DoraMetrics/DoraCards/ChangeFailureRateCard.tsx]: fixed

**Review by jayantbh — APPROVED** [review]

**Pull request merged**